src/vcpkg/vcpkgpaths.cpp

```diff
@@ -497,25 +498,63 @@ namespace vcpkg
         });
     }
 
-    const std::map<std::string, std::string>& VcpkgPaths::get_cmake_script_hashes() const
+    ContentAndHash::ContentAndHash(std::string&& ctnt, Hash::Algorithm algo) : content_(ctnt)
     {
-        return m_pimpl->cmake_script_hashes.get_lazy([this]() -> std::map<std::string, std::string> {
+        hash_ = Hash::get_string_hash(content(), algo);
+    }
+
+    const std::map<std::string, ContentAndHash>& VcpkgPaths::get_cmake_script_content_and_hashes() const
+    {
+        return m_pimpl->cmake_script_content_and_hashes.get_lazy([this]() {
             auto& fs = this->get_filesystem();
-            std::map<std::string, std::string> helpers;
+            std::map<std::string, ContentAndHash> helpers;
             auto files = fs.get_files_non_recursive(this->scripts / fs::u8path("cmake"));
             for (auto&& file : files)
             {
-                helpers.emplace(fs::u8string(file.stem()),
-                                Hash::get_file_hash(VCPKG_LINE_INFO, fs, file, Hash::Algorithm::Sha1));
+                auto content = fs.read_contents(file, VCPKG_LINE_INFO);
+                helpers.emplace(fs::u8string(file.stem()), ContentAndHash(std::move(content), Hash::Algorithm::Sha1));
             }
             return helpers;
         });
     }
 
+    const std::map<std::string, ContentAndHash>& VcpkgPaths::get_buildsystem_script_content_and_hashes() const
+    {
+        return m_pimpl->buildsystem_script_content_and_hashes.get_lazy([this]() {
+            auto& fs = this->get_filesystem();
+            std::map<std::string, ContentAndHash> buildsystem_scripts;
+            auto buildsystem_directory = this->scripts / fs::u8path("buildsystems");
+            auto buildsystem_files = fs.get_files_recursive(buildsystem_directory);
+            for (auto&& file : buildsystem_files)
+            {
+                if (fs.is_directory(file))
+                {
+                    continue;
+                }
+                auto relative_file = this->scripts.lexically_relative(file);
```

Comment:
lexically_relative doesn't exist in experimental filesystem so this will blow up on some of our supported platforms.